File: backend/src/main/java/com/nextivis/nishopping/domain/service/ProductService.java
package com.nextivis.nishopping.domain.service;

import com.nextivis.nishopping.domain.entity.Product;
import com.nextivis.nishopping.domain.repository.ProductRepository;
import com.nextivis.nishopping.domain.service.helper.RandomHelper;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;
import java.util.List;

@Service
@AllArgsConstructor
public class ProductService {

    private final ProductRepository productRepository;
    private final RandomHelper randomHelper;

    public int createProduct(Product product) {
        String newPid = randomHelper.getUUID();
        LocalDateTime releaseDate = LocalDateTime.now().truncatedTo(ChronoUnit.SECONDS);
        // TODO : duplicatePID check
        product.setPid(newPid);
        product.setReleaseDate(releaseDate);
        return productRepository.create(product);
    }

    public List<Product> getAllProduct() {
        return productRepository.findAll();
    }

    public Product getProductByPid(String pid) {
        return productRepository.findByPid(pid);
    }

    public List<Product> getProductByGenre(String genre) {
        return productRepository.findByGenre(genre);
    }

    public int updateProduct(Product product) {
        return productRepository.updateProduct(product);
    }
}

File: frontend/src/service/productService.js
import { defaultGet, defaultPost } from './apiService';

const createProduct = async (productInfo) => {
  return defaultPost('/product/create', productInfo);
};

// Update의 경우, 단순히 productInfo를 날려도 되는가?
const updateProduct = async (productInfo) => {
  return defaultPost('/product/update', productInfo);
};

const findAllProduct = async (order) => {
  return defaultGet(`/product/find/all?order=${order}`);
};

const findById = async (pid) => {
  return defaultGet(`/product/find?pid=${pid}`);
};

const findByGenre = async (genre) => {
  return defaultGet(`/product/find?genre=${genre}`);
};

export {
  createProduct,
  updateProduct,
  findAllProduct,
  findById,
  findByGenre,
};
